### ISSUE TITLE
Failed to build on raspbian 32 bits for RPi400

### ISSUE THREAD

**Issue by lribeiro** [ClickHouse/clickhouse-go#519]
### Issue description
Code compilation error due to overflow on 32bits

Backtrace points to
https://github.com/ClickHouse/clickhouse-go/blob/v2/lib/column/lowcardinality.go#L219

`case len(col.append.index) < math.MaxUint32:`

My guess would be len(...) returns int which in a 32bit env can't support math.MaxUint32 without overflow.

### Example code
```go
Just build it on a rpi 32 bits OS
```

### Error log
```
# github.com/ClickHouse/clickhouse-go/v2/lib/column
../go/pkg/mod/github.com/!click!house/clickhouse-go/v2@v2.0.12/lib/column/lowcardinality.go:219:29: constant 4294967295 overflows int
make: *** [Makefile:37: linux_arm] Error 1
make: *** [linux_pi] Error 2
```

### Configuration
*CPU*
...
processor       : 3
BogoMIPS        : 108.00
Features        : fp asimd evtstrm crc32 cpuid
CPU implementer : 0x41
CPU architecture: 8
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

Hardware        : BCM2835
Revision        : c03130
Serial          : 1000000066e5fb99
Model           : Raspberry Pi 400 Rev 1.0

*OS:*
raspbian bullseye
With Kernel

`Linux raspberrypi4 5.10.63-v8+ #1488 SMP PREEMPT Thu Nov 18 16:16:16 GMT 2021 aarch64 GNU/Linux`
The CPU actually supports aarch64 but the OS is running 32 bits sw

*Interface:*
`database/sql`

*Driver version:*
`v2@v2.0.12`

*Go version:* run `go version` in your console
`go version go1.17.6 linux/arm`

*ClickHouse Server version:*
Didn't get that far :)

**Comment by genzgd** [comment]
You are absolutely correct.  We will merge the fix shortly

**Comment by gingerwizard** [comment]
Fixed by #563
